### PR TITLE
hardening/1208_fix_automated_build_stuck_filelock_denied

### PR DESCRIPTION
**Pull request by cesarjorgemartinez**

### CODE REVIEW

**Comment by cesarjorgemartinez** [comment]
- Test build in https://hub.docker.com/r/telefonicaiot/fiware-cygnus/builds/bbkqfxcuuxysx8lt8vgpc5k/

- Before merge this PR we need to add a commit to change GIT_URL_CYGNUS/GIT_REV_CYGNUS

- We need to change all documentation referenced: change to JDK 1.7, maven version packages changed, etc.

**Comment by cesarjorgemartinez** [comment]
First suggested changes:
DOCS
```

git diff doc/cygnus-common/installation_and_administration_guide/install_from_sources.md
diff --git a/doc/cygnus-common/installation_and_administration_guide/install_from_sources.md b/doc/cygnus-common/installation_and_administration_guide/install_from_sources.md
index 6df9778..3dbf5c6 100644
--- a/doc/cygnus-common/installation_and_administration_guide/install_from_sources.md
+++ b/doc/cygnus-common/installation_and_administration_guide/install_from_sources.md
@@ -14,13 +14,13 @@ Content:
 ##<a name="section1"></a>Prerequisites
 Maven (and thus Java SDK, since Maven is a Java tool) is needed in order to install cygnus-common.
 
-In order to install Java SDK (not JRE), just type (CentOS machines):
+In order to install Java SDK (not JRE), minimum version 1.7, just type (CentOS machines):
 
-    $ yum install java-1.6.0-openjdk-devel
+    $ yum install java-1.7.0-openjdk-devel
 
 Remember to export the JAVA_HOME environment variable. In the case of using `yum install` as shown above, it would be:
 
-    $ export JAVA_HOME=/usr/lib/jvm/java-1.6.0-openjdk.x86_64
+    $ export JAVA_HOME=/usr/lib/jvm/java-1.7.0-openjdk.x86_64
 
 In order to do it permanently, edit `/root/.bash_profile` (`root` user) or `/etc/profile` (other users).
 
@@ -115,7 +115,7 @@ The installation is completed by copying the `cygnus-flume-ng` script into `APAC
 ###<a name="section4.4"></a>Known issues
 It may happen while compiling `cygnus-common` the Maven JVM has not enough memory. This can be changed as detailed at the [Maven official documentation](https://cwiki.apache.org/confluen
 
-    $ export MAVEN_OPTS="-Xmx512m -XX:MaxPermSize=128m"
+    $ export MAVEN_OPTS="-Xmx512m -XX:MaxPermSize=128m -Dfile.encoding=UTF-8"
 
 [Top](#top)

aaaaaaaaaaaaaaaaaaaaa

git diff doc/cygnus-common/installation_and_administration_guide/install_with_docker.md
diff --git a/doc/cygnus-common/installation_and_administration_guide/install_with_docker.md b/doc/cygnus-common/installation_and_administration_guide/install_with_docker.md
index 63e5835..05e7c90 100644
--- a/doc/cygnus-common/installation_and_administration_guide/install_with_docker.md
+++ b/doc/cygnus-common/installation_and_administration_guide/install_with_docker.md
@@ -69,7 +69,7 @@ Start a container for this image by typing in a terminal:
 Immediately after, you will start seeing cygnus-common logging traces:
 
 ```
-+ exec /usr/lib/jvm/java-1.6.0/bin/java -Xmx20m -Dflume.root.logger=INFO,console -cp '/opt/apache-flume/conf:/opt/apache-flume/lib/*:/opt/apache-flume/plugins.d/cygnus/lib/*:/opt/apache-
++ exec /usr/lib/jvm/java-1.7.0/bin/java -Xmx20m -Dflume.root.logger=INFO,console -cp '/opt/apache-flume/conf:/opt/apache-flume/lib/*:/opt/apache-flume/plugins.d/cygnus/lib/*:/opt/apache-
 SLF4J: Class path contains multiple SLF4J bindings.
 SLF4J: Found binding in [jar:file:/opt/apache-flume/lib/slf4j-log4j12-1.6.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
 SLF4J: Found binding in [jar:file:/opt/apache-flume/plugins.d/cygnus/libext/cygnus-common-1.0.0_SNAPSHOT-jar-with-dependencies.jar!/org/slf4j/impl/StaticLoggerBinder.class]
aaa


git diff doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md
diff --git a/doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md b/doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md
index 0a41ba7..dfe70c3 100644
--- a/doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md
+++ b/doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md
@@ -69,7 +69,7 @@ Start a container for this image by typing in a terminal:
 Immediately after, you will start seeing cygnus-ngsi logging traces:
 
 ```
-+ exec /usr/lib/jvm/java-1.6.0/bin/java -Xmx20m -Dflume.root.logger=INFO,console -cp '/opt/apache-flume/conf:/opt/apache-flume/lib/*:/opt/apache-flume/plugins.d/cygnus/lib/*:/opt/apache-
++ exec /usr/bin/java -Xmx20m -Dflume.root.logger=INFO,console -cp '/opt/apache-flume/conf:/opt/apache-flume/lib/*:/opt/apache-flume/plugins.d/cygnus/lib/*:/opt/apache-flume/plugins.d/cyg
 SLF4J: Class path contains multiple SLF4J bindings.
 SLF4J: Found binding in [jar:file:/opt/apache-flume/lib/slf4j-log4j12-1.6.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
 SLF4J: Found binding in [jar:file:/opt/apache-flume/plugins.d/cygnus/lib/cygnus-ngsi-0.13.0_SNAPSHOT-jar-with-dependencies.jar!/org/slf4j/impl/StaticLoggerBinder.class]
aaaaaaaaa




git diff doc/cygnus-ngsi/quick_start_guide.md
diff --git a/doc/cygnus-ngsi/quick_start_guide.md b/doc/cygnus-ngsi/quick_start_guide.md
index 47f30ab..dee4a6d 100644
--- a/doc/cygnus-ngsi/quick_start_guide.md
+++ b/doc/cygnus-ngsi/quick_start_guide.md
@@ -17,16 +17,16 @@ $ sudo yum install cygnus
 
 **IMPORTANT NOTE:**
 
-Cygnus, as it is based on Apache Flume, needs Java SDK is installed. If not yet installed, do it right now:
+Cygnus, as it is based on Apache Flume, needs Java SDK is installed, minimum version 1.7. If not yet installed, do it right now:
 
 ```
-$ sudo yum install java-1.6.0-openjdk-devel
+$ sudo yum install java-1.7.0-openjdk-devel
 ```
 
 Remember to export the JAVA_HOME environment variable. In the case you have installed Java SDK right now as shown above, this would be as:
 
 ```
-$ export JAVA_HOME=/usr/lib/jvm/java-1.6.0-openjdk.x86_64
+$ export JAVA_HOME=/usr/lib/jvm/java-1.7.0-openjdk.x86_64
 ```
 
 In order to do it permanently, edit `/root/.bash_profile` (root user) or `/etc/profile` (other users).
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa



git diff doc/cygnus-twitter/installation_and_administration_guide/install_with_docker.md
diff --git a/doc/cygnus-twitter/installation_and_administration_guide/install_with_docker.md b/doc/cygnus-twitter/installation_and_administration_guide/install_with_docker.md
index 730f90f..439691b 100644
--- a/doc/cygnus-twitter/installation_and_administration_guide/install_with_docker.md
+++ b/doc/cygnus-twitter/installation_and_administration_guide/install_with_docker.md
@@ -69,7 +69,7 @@ Start a container for this image by typing in a terminal:
 Immediatelly after, you will start seeing cygnus-twitter logging traces:
 
 ```
-+ exec /usr/lib/jvm/java-1.6.0/bin/java -Xmx20m -Dflume.root.logger=INFO,console -cp '/opt/apache-flume/conf:/opt/apache-flume/lib/*:/opt/apache-flume/plugins.d/cygnus/lib/*:/opt/apache-
++ exec /usr/lib/jvm/java-1.7.0/bin/java -Xmx20m -Dflume.root.logger=INFO,console -cp '/opt/apache-flume/conf:/opt/apache-flume/lib/*:/opt/apache-flume/plugins.d/cygnus/lib/*:/opt/apache-
 SLF4J: Class path contains multiple SLF4J bindings.
 SLF4J: Found binding in [jar:file:/opt/apache-flume/lib/slf4j-log4j12-1.6.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
 SLF4J: Found binding in [jar:file:/opt/apache-flume/plugins.d/cygnus/lib/cygnus-twitter-0.1.0_SNAPSHOT-jar-with-dependencies.jar!/org/slf4j/impl/StaticLoggerBinder.class]
```

File: docker/cygnus-common/Dockerfile [hunk 1]
```diff
@@ -45,18 +46,19 @@ ENV FLUME_TGZ "apache-flume-${FLUME_VER}-bin.tar.gz"
 ENV FLUME_URL "http://archive.apache.org/dist/flume/${FLUME_VER}/${FLUME_TGZ}"
 ENV FLUME_HOME "/opt/apache-flume"
 
-ENV JAVA_HOME /usr/lib/jvm/java-1.6.0
+ENV JAVA_VERSION=1.7.0
```
Comment (frbattid) [inline docker/cygnus-common/Dockerfile]: Is this really working, by using the `=`? It seems the environment variables in docker have the form `ENV name value`, without the equals character.

File: docker/cygnus-common/Dockerfile [hunk 2]
```diff
@@ -45,18 +46,19 @@ ENV FLUME_TGZ "apache-flume-${FLUME_VER}-bin.tar.gz"
...
 # Install
-RUN yum -y install tar nc which git java-1.6.0-openjdk-devel && \
+RUN yum -y install tar nc which git java-${JAVA_VERSION}-openjdk-devel && \
     yum clean all && \
+    export JAVA_HOME=/usr/lib/jvm/java-${JAVA_VERSION} && \
     curl --remote-name --location --insecure --silent --show-error "${MVN_URL}" && \
     tar xzvf ${MVN_TGZ} && \
     mv apache-maven-${MVN_VER} ${MVN_HOME} && \
-    export MAVEN_OPTS="-Xmx512m -XX:MaxPermSize=128m" && \
+    export MAVEN_OPTS="-Xmx512m -XX:MaxPermSize=128m -Dfile.encoding=UTF-8 -Dproject.build.sourceEncoding=UTF-8" && \
```
Comment (frbattid) [inline docker/cygnus-common/Dockerfile]: As discussed offline, I think this wont affect the JVM but only Maven.

`-Dfile.encoding=UTF-8` must be added to the starting command within `cygnus-entrypoint.sh` instead.

File: docker/cygnus-ngsi/Dockerfile [hunk 3]
```diff
@@ -36,77 +36,95 @@ ENV CYGNUS_MYSQL_PASS "mysql"
...
+ENV MVN_VER "3.3.9"
 ENV MVN_HOME "/opt/apache-maven"
 
 ENV FLUME_VER "1.4.0"
-ENV FLUME_TGZ "apache-flume-${FLUME_VER}-bin.tar.gz"
-ENV FLUME_URL "http://archive.apache.org/dist/flume/${FLUME_VER}/${FLUME_TGZ}"
 ENV FLUME_HOME "/opt/apache-flume"
 
-ENV JAVA_HOME /usr/lib/jvm/java-1.6.0
+ENV JAVA_VERSION=1.7.0
```
Comment (frbattid) [inline docker/cygnus-ngsi/Dockerfile]: Is this really working, by using the `=`? It seems the environment variables in docker have the form `ENV name value`, without the equals character.

File: docker/cygnus-ngsi/Dockerfile [hunk 4]
```diff
@@ -36,77 +36,95 @@ ENV CYGNUS_MYSQL_PASS "mysql"
...
 
 # Install
 RUN \
     # Add Cygnus user
     adduser ${CYGNUS_USER} && \
-    yum -y install nc java-1.6.0-openjdk-devel git && \
+    yum -y install nc java-${JAVA_VERSION}-openjdk-devel git && \
+    export JAVA_HOME=/usr/lib/jvm/java-${JAVA_VERSION} && \
+    export MAVEN_OPTS="-Xmx512m -XX:MaxPermSize=128m -Dfile.encoding=UTF-8 -Dproject.build.sourceEncoding=UTF-8 -Dmaven.compiler.useIncrementalCompilation=false -DdependencyLocationsEnabled=false -XX:+TieredCompilation -XX:TieredStopAtLevel=1" && \
```
Comment (frbattid) [inline docker/cygnus-ngsi/Dockerfile]: As discussed offline, I think this wont affect the JVM but only Maven.

`-Dfile.encoding=UTF-8` must be added to the starting command within `cygnus-entrypoint.sh` instead.

File: docker/cygnus-twitter/Dockerfile [hunk 5]
```diff
@@ -43,18 +44,19 @@ ENV FLUME_TGZ "apache-flume-${FLUME_VER}-bin.tar.gz"
...
 # Install
-RUN yum -y install tar nc which git java-1.6.0-openjdk-devel && \
+RUN yum -y install tar nc which git java-${JAVA_VERSION}-openjdk-devel && \
     yum clean all && \
+    export JAVA_HOME=/usr/lib/jvm/java-${JAVA_VERSION} && \
     curl --remote-name --location --insecure --silent --show-error "${MVN_URL}" && \
     tar xzvf ${MVN_TGZ} && \
     mv apache-maven-${MVN_VER} ${MVN_HOME} && \
-    export MAVEN_OPTS="-Xmx512m -XX:MaxPermSize=128m" && \
+    export MAVEN_OPTS="-Xmx512m -XX:MaxPermSize=128m -Dfile.encoding=UTF-8 -Dproject.build.sourceEncoding=UTF-8" && \
```
Comment (frbattid) [inline docker/cygnus-twitter/Dockerfile]: As discussed offline, I think this wont affect the JVM but only Maven.

`-Dfile.encoding=UTF-8` must be added to the starting command within `cygnus-entrypoint.sh` instead.

File: docker/cygnus-twitter/Dockerfile [hunk 6]
```diff
@@ -43,18 +44,19 @@ ENV FLUME_TGZ "apache-flume-${FLUME_VER}-bin.tar.gz"
 ENV FLUME_URL "http://archive.apache.org/dist/flume/${FLUME_VER}/${FLUME_TGZ}"
 ENV FLUME_HOME "/opt/apache-flume"
 
-ENV JAVA_HOME /usr/lib/jvm/java-1.6.0
+ENV JAVA_VERSION=1.7.0
```
Comment (frbattid) [inline docker/cygnus-twitter/Dockerfile]: Is this really working, by using the `=`? It seems the environment variables in docker have the form `ENV name value`, without the equals character.

File: docker/cygnus-ngsi/Dockerfile (hunk 3 above)
Comment (cesarjorgemartinez) [inline docker/cygnus-ngsi/Dockerfile]: Are two different nomenclatures, but we use with spaces, changed

File: docker/cygnus-ngsi/Dockerfile (hunk 4 above)
Comment (cesarjorgemartinez) [inline docker/cygnus-ngsi/Dockerfile]: Adding the parameter also in the entrypoint to ensure

File: docker/cygnus-twitter/Dockerfile (hunk 5 above)
Comment (cesarjorgemartinez) [inline docker/cygnus-twitter/Dockerfile]: Adding the parameter also in the entrypoint to ensure

**Comment by pcoello25** [comment]
I've done checkout of your branch in my machine for testing: Successfully installed and less time for the process (12 min -> 3-4 min). It shows warning in the installation process such:
```
"Oct 19, 2016 6:27:40 AM com.sun.java.util.jar.pack.Utils$Pack200Logger warning
WARNING: Passing class file uncompressed due to unrecognized attribute: scala/util/parsing/json/...."
```
but itdoesn't stop the process. 

I've tested a Cygnus running following the guide and all works properly :+1:

**Comment by frbattid** [comment]
LGTM (because @pcoello25 LGTMs ;))

**Pull request merged**